File: abc_311_a.cpp
/**
  *  programmer:  Zama
*    created: 22.07.2023 20:59:35
**/

#include <bits/stdc++.h>

using namespace std;

int main() {
  int N;
  string S;
  cin >> N >> S;

  bool a = 0, b = 0, c = 0;
  for(int i = 0; i < N; i++) {
    if (S[i] == 'A') a = 1;
    if (S[i] == 'B') b = 1;
    if (S[i] == 'C') c = 1;
    if (a && b && c == 1) {
      cout << i + 1 << endl;
      return 0;
    }
  }

}

File: abc_311_a.py
##########################################
#    programmer:  Zama
#    created: Oct 09, 2023 - 00:30:03
##########################################

import numpy as np
import math
import string
import sys

n = input()
s = input()

flag = [0, 0, 0]
for i, e in enumerate(s):
    if (e == "A"):
        flag[0] = 1
    if (e == "B"):
        flag[1] = 1
    if (e == "C"):
        flag[2] = 1
    if (flag[0] * flag[1] * flag[2] == 1):
        print(i+1)
        exit()